main/src/resources/benchmark/BenchmarkSet.flix

```diff
@@ -0,0 +1,204 @@
+    @benchmark
+    pub def benchmark20(): Bool = Set.isSubsetOf(Set.range(0, 99), Set.range(0, 100))
+
+    ///
+    /// create a set of integers, increment each integer by one and compute the size.
+    ///
+    @benchmark
+    pub def benchmark21(): Bool = Set.size(Set.map(x -> x + 1, Set.range(0, 100))) == 100
+
+    ///
+    /// create a set of integers, partition it into even and odd numbers and compute the size of the first partition.
+    ///
+    @benchmark
+    pub def benchmark22(): Bool =
+        let partition = Set.partition(x -> x % 2 == 0, Set.range(0, 100));
+        let (fst, _) = partition;
+        Set.size(fst) == 50
+
+    ///
+    /// create a set of integers and compute its sum via reduceLeft.
+    ///
+    @benchmark
+    pub def benchmark23(): Bool = Set.reduceLeft((x, y) -> x + y, Set.range(0, 100)) == Some(4950)
+
+    ///
+    /// create a set of integers and compute its sum via reduceRight.
+    ///
+    @benchmark
+    pub def benchmark24(): Bool = Set.reduceRight((x, y) -> x + y, Set.range(0, 100)) == Some(4950)
+
+    ///
+    /// create a set of integers and replace a low-valued element with a high-valued element.
+    ///
+    @benchmark
+    pub def benchmark25(): Bool = Set.memberOf(100, Set.replace(0, 100, Set.range(0, 100)))
+
+    ///
+    /// create a set of integers and replace a mid-valued element with a high-valued element.
+    ///
+    @benchmark
+    pub def benchmark26(): Bool = Set.memberOf(100, Set.replace(50, 100, Set.range(0, 100)))
+
+    ///
+    /// create a set of integers and replace a high-valued element with a low valued-element.
+    ///
+    @benchmark
+    pub def benchmark27(): Bool = Set.memberOf(-1, Set.replace(99, -1, Set.range(0, 100)))
+
+    ///
+    /// create a set of 4 integers and compute the size of its subsets.
+    ///
+    @benchmark
+    pub def benchmark28(): Bool = Set.size(Set.subsets(Set.range(0, 4))) == 16
+
+    ///
+    /// create a set of 6 integers and compute the size of its subsets.
+    ///
+    @benchmark
+    pub def benchmark29(): Bool = Set.size(Set.subsets(Set.range(0, 6))) == 64
+
+    ///
+    /// create two sets of integers and compute the size of their union.
+    ///
+    @benchmark
+    pub def benchmark30(): Bool = Set.size(Set.union(Set.range(0, 50), Set.range(25, 50))) == 50
+
+    ///
+    /// create two sets of integers and compute the size of their union.
+    ///
+    @benchmark
+    pub def benchmark31(): Bool = Set.size(Set.union(Set.range(0, 75), Set.range(50, 100))) == 100
+
+    ///
+    /// create a set of integers, compute all its subsets, flatten (union) them and compute the size of the resulting set.
+    ///
+    @benchmark
+    pub def benchmark32(): Bool = Set.size(Set.flatten(Set.subsets(Set.range(0, 4)))) == 4
+
+    ///
+    /// create a set of integers, convert it to a list and compute the length of the list.
+    ///
+    @benchmark
+    pub def benchmark33(): Bool = List.length(Set.toList(Set.range(0, 100))) == 100
+
```

Comment:
Do all of these benchmarks run in reasonable time compared to e.g. BenchmarkList?

Reply:
No, not currently. I'll adjust them.